### PR TITLE
Harish

### PR DESCRIPTION
**Pull request by devarshjoshi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   },
   "dependencies": {
     "axios": "^0.17.1",
+    "child-process-promise": "^2.2.1",
```
Comment (devarshjoshi) [inline package.json]: please remove

File: src/api/process_ipfs.js [hunk 2]
```diff
@@ -12,24 +12,41 @@ var log = require('electron-log');
...
+
 var ipfs = null;
 
+var isIpfsReady = false;
+
+
+
+// Keep updating the global variable 'isIpfsReady'
+
+function updateIPFSAvailability(){
```
Comment (devarshjoshi) [inline src/api/process_ipfs.js]: please remove

File: src/app.js [hunk 3]
```diff
@@ -67,6 +69,9 @@ var start = function start(){
         if (process.platform !== 'darwin') {
             app.quit();
         }
+
+        // Testing quit functionality on Mac
```
Comment (devarshjoshi) [inline src/app.js:73]: confirm with rak if it works.

File: src/pages.js [hunk 4]
```diff
@@ -61,19 +63,28 @@ var getTray = (function(){
...
-        y: 180,
-        width: 1147,
-        height: 849
+        center: true,
+        width: 0.6*electronScreen.width,
+        height: 0.7*electronScreen.height
 
     });
 
+    //console.log("Window Dimensions "+ electronScreen.getPrimaryDisplay().size.height);
```
Comment (devarshjoshi) [inline src/pages.js]: do we need this comments here ?

File: src/shared/actions/ipfs.js [hunk 5]
```diff
@@ -76,7 +76,16 @@ var start = exports.start = createAliasedAction( "start", function(){
 	return function(dispatch){
 		dispatch(logOutput('starting..'));
 		dispatch(logOutput('Starting IPFS...'));
+
 		ipfs.startIpfs((message)=> dispatch(logOutput(message)));
+
+		var cust = function(msg){
```
Comment (devarshjoshi) [inline src/shared/actions/ipfs.js]: please remove

File: src/app.js (hunk 3 above)
Comment (harishrongala) [reply src/app.js:73]: Seems to be working

Comment (harishrongala) [reply src/app.js:73]: I would like to leave this unchanged. In the future, If any Mac-specific issue arises, we could come back to this piece of code.

**Review by devarshjoshi — COMMENTED** [review]
done

**Pull request merged**